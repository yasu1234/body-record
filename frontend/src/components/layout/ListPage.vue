<script setup>
import { ref, onMounted, watch } from "vue";

import Paginate from "vuejs-paginate-next";

const pageCount = ref(0);

const props = defineProps({
  pageCount: Number,
});

onMounted(() => {
  pageCount.value = props.pageCount;
});

watch(props, () => {
  pageCount.value = props.pageCount;
});

const emits = defineEmits(["changePage"]);

const clickCallback = function (pageNum) {
  emits("changePage", pageNum);
};
</script>

<template>
  <Paginate
    :page-count="pageCount"
    :page-range="3"
    :margin-pages="2"
    :click-handler="clickCallback"
    :prev-text="'<'"
    :next-text="'>'"
  />
</template>

<style scoped>
.pagination {
  display: flex;
  justify-content: center;
  margin-top: 20px;
  list-style: none;
}

:deep .page-item {
  color: #333;
  text-decoration: none;
  transition: background-color 0.3s;
  border: 1px solid #ddd;
  margin: 0 4px;
  width: 30px;
  height: 30px;
  padding: 20px;
}

:deep .page-link {
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0;
  margin: 0;
}

:deep .page-item.active {
  background-color: #ffa500;
  color: white;
  border: 1px solid #ffa500;
}

:deep .page-item.disabled {
  cursor: not-allowed;
  color: #332;
}

:deep .page-item:hover:not(.disabled):not(.active) {
  background-color: #f5c977;
}

:deep .page-link.disabled {
  color: #ccc;
  cursor: not-allowed;
}
</style>
