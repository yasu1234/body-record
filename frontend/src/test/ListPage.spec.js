import { shallowMount } from '@vue/test-utils'
import ListPage from '@/components/layout/ListPage.vue'

jest.mock('axios')
jest.mock('vuejs-paginate-next')
jest.mock('vue-router', () => {
  const router = {
    push: jest.fn(),
    pathname: '/',
    route: '',
    asPath: '',
    query: {},
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
  };
  return {
    useRouter: jest.fn().mockReturnValue(router),
  };
});

describe('TabItem.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(ListPage, {
      global: {
        stubs: ["router-link", "router-view"]
      }
    })
  })

  it('SnapShotテスト(合計1ページ)', () => {
    wrapper = shallowMount(ListPage, {
        props: {
            pageCount: 1
        }
    })

    expect(wrapper.element).toMatchSnapshot();
  })

  it('SnapShotテスト(合計10ページ)', () => {
    wrapper = shallowMount(ListPage, {
        props: {
            pageCount: 10
        }
    })

    expect(wrapper.element).toMatchSnapshot();
  })
})
