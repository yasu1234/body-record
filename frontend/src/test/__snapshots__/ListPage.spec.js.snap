// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabItem.vue SnapShotテスト(合計1ページ) 1`] = `
<anonymous-stub
  activeclass="active"
  breakviewtext="…"
  clickhandler="[Function]"
  containerclass="pagination"
  disabledclass="disabled"
  firstbuttontext="First"
  firstlastbutton="false"
  hideprevnext="false"
  initialpage="1"
  lastbuttontext="Last"
  marginpages="2"
  nextclass="page-item"
  nextlinkclass="page-link"
  nexttext=">"
  nolisurround="false"
  pageclass="page-item"
  pagecount="1"
  pagelinkclass="page-link"
  pagerange="3"
  prevclass="page-item"
  prevlinkclass="page-link"
  prevtext="<"
/>
`;

exports[`TabItem.vue SnapShotテスト(合計10ページ) 1`] = `
<anonymous-stub
  activeclass="active"
  breakviewtext="…"
  clickhandler="[Function]"
  containerclass="pagination"
  disabledclass="disabled"
  firstbuttontext="First"
  firstlastbutton="false"
  hideprevnext="false"
  initialpage="1"
  lastbuttontext="Last"
  marginpages="2"
  nextclass="page-item"
  nextlinkclass="page-link"
  nexttext=">"
  nolisurround="false"
  pageclass="page-item"
  pagecount="10"
  pagelinkclass="page-link"
  pagerange="3"
  prevclass="page-item"
  prevlinkclass="page-link"
  prevtext="<"
/>
`;
